src/decision/Criteria/DecisionPreferences.jl

```diff
@@ -140,7 +140,9 @@ function rank_by_index(
     end
 
     # Recreate preferences, removing criteria that are constant for this scenario
-    _dp = filter_criteria(dp, is_const)
+    if !all(.!is_const)
+        _dp = filter_criteria(dp, is_const)
+    end
 
     # Assess decision matrix only using valid (non-constant) criteria
     res = solve(_dp, dm[criteria=.!is_const], method)
```

Comment:
If it doen't enter in the if `_dp` is not defined. Also, this was causing this error when I tried to run: `ERROR: UndefVarError: `_dp` not defined in local scope Suggestion: check for an assignment to a local variable that shadows
a global of the same name.`
You could just reassign `dp`
```suggestion
    if !all(.!is_const)
        dp = filter_criteria(dp, is_const)
    end

    # Assess decision matrix only using valid (non-constant) criteria
    res = solve(dp, dm[criteria=.!is_const], method)
```